feat: code polish on classifier

yuluo-yx opened the pull request:
1. optimize  semantic-router/util/classification

Xunzhuo commented on src/semantic-router/pkg/utils/classification/classifier.go:
```diff
@@ -507,6 +497,13 @@ func (c *Classifier) SelectBestModelForCategory(categoryName string) string {
 	return bestModel
 }
 
+// forEachModelScore 遍历 category 的 ModelScores 并对每个元素执行回调
```
plz update the comment

yuluo-yx replied:
done

Xunzhuo reviewed (commented):
can you give it a simple conclusion of what you optimized for classifier?

yuluo-yx commented:
> can you give it a simple conclusion of what you optimized for classifier?

sure, 

1. i delete the contains func, which the go slices library includes.
2. extracted the `forEachModelScore` func because the line `for _, modelScore := range cat.ModelScores {` appears four times in the file, which is not concise.
3. extracted `updateBestModel` func.

Xunzhuo commented:
and add ut around your code polish in this PR to make sure things go in expectations.

Xunzhuo commented on src/semantic-router/pkg/utils/classification/classifier_test.go:
```diff
@@ -406,3 +406,49 @@ var _ = Describe("PIIClassification", func() {
+
+	c := &Classifier{}
+	cat := &config.Category{
+		ModelScores: []config.ModelScore{
```
the order is exactly the same as you check in line 445, this makes the test untrustable, can you make it unordered?

yuluo-yx replied:
sorry, i don`t undersatnd.

i used the append func here to append models, and the appended content should be the grid.

Xunzhuo replied:
Aha, I got u sorry I think misunderstood it, plz go ahead
